src/index.ts

```diff
@@ -132,15 +134,59 @@ function addSegmentFromDirectionsResponse(previousLngLat: LngLat, e: MapMouseEve
     });
 }
 
-function addSegmentFromStraightLine(previousLngLat: LngLat, e: MapMouseEvent): void {
-  const newSegment = nextSegmentService.segmentFromStraightLine(previousLngLat, e.lngLat);
+function addSegmentFromStraightLine(previousLngLat: LngLat, lngLat: LngLat, animate = true): void {
+  const newSegment = nextSegmentService.segmentFromStraightLine(previousLngLat, lngLat);
 
-  animationService.animateSegment(newSegment);
-  const marker = addMarker(e.lngLat, false);
+  if (animate) animationService.animateSegment(newSegment);
+  const marker = addMarker(lngLat, false);
   currentRun.addSegment(newSegment, marker);
   updateLengthElement();
 }
 
+function runToJson(run: CurrentRun): string {
+  if (run === undefined) return "{}";
+  let runJSON: {[name:string]: any} = {
```

Comment:
minor, but may be good to use an interface for this since we use `runJSON.segments.push(...` down in the loop and I'd hate for this to get out of sync with how it's used

Reply:
What do you mean?

Reply:
ah, apologies for not being super clear, the simple key:value type you've got for `runJSON` should be fine, nevermind!